python/src/main.cpp

```diff
@@ -257,40 +264,14 @@ PYBIND11_MODULE(kp, m)
                          "size {} dtype {}",
                          flatdata.size(),
                          std::string(py::str(flatdata.dtype())));
-            if (flatdata.dtype().is(py::dtype::of<std::float_t>())) {
-                return self.tensor(info.ptr,
-                                   flatdata.size(),
-                                   sizeof(float),
-                                   kp::Tensor::TensorDataTypes::eFloat,
-                                   tensor_type);
-            } else if (flatdata.dtype().is(py::dtype::of<std::uint32_t>())) {
-                return self.tensor(info.ptr,
-                                   flatdata.size(),
-                                   sizeof(uint32_t),
-                                   kp::Tensor::TensorDataTypes::eUnsignedInt,
-                                   tensor_type);
-            } else if (flatdata.dtype().is(py::dtype::of<std::int32_t>())) {
-                return self.tensor(info.ptr,
-                                   flatdata.size(),
-                                   sizeof(int32_t),
-                                   kp::Tensor::TensorDataTypes::eInt,
-                                   tensor_type);
-            } else if (flatdata.dtype().is(py::dtype::of<std::double_t>())) {
-                return self.tensor(info.ptr,
-                                   flatdata.size(),
-                                   sizeof(double),
-                                   kp::Tensor::TensorDataTypes::eDouble,
-                                   tensor_type);
-            } else if (flatdata.dtype().is(py::dtype::of<bool>())) {
-                return self.tensor(info.ptr,
-                                   flatdata.size(),
-                                   sizeof(bool),
-                                   kp::Tensor::TensorDataTypes::eBool,
-                                   tensor_type);
-            } else {
-                throw std::runtime_error(
-                  "Kompute Python no valid dtype supported");
-            }
+
+            std::shared_ptr<ABCTypeContainer> typeContainer =
+              std::make_shared<PyTypeContainer>(flatdata.dtype());
```

Comment:
Why does type need to be sharedptr? Let's reduce complexity on type info

Reply:
changed to regular pointer